src/rooms/rooms.gateway.ts

```diff
@@ -0,0 +1,28 @@
+import { WebSocketGateway, WebSocketServer } from '@nestjs/websockets';
+import { Server } from 'socket.io';
+import { ResRoomDto } from './dtos/responses/room-response.dto';
+import { Room } from './entities/rooms.entity';
+
+@WebSocketGateway({
+  cors: {
+    origin: process.env.ALLOWED_ORIGINS?.split(','),
+    credentials: true,
+  },
+  namespace: '/rooms',
+})
+export class RoomsGateway {
```

Comment:
게이트웨이는 입출력 포트(controller와 비슷한 역할)을 합니다.
service를 보니 RoomGateWay를 그대로 DI 받는것으로 확인했어요.
service가 controller를 DI 받는다면 어색하다고 느껴져요.

당장에 socket server에 접근하기 위해서 WebSocketGateWay 데코레이터를 사용하니 클래스 명 자체는 RoomsGateway로 하는것에는 동의하는데요.
이벤트 퍼블리싱만을 위한 port interface를 따로 두어 하는건 어떻게 생각하실까요??
지금처럼 구현체 자체를 의존한다면 불필요한 인터페이스까지 접근이 가능해집니다.(DIP)

이벤트 퍼블리셔만 똑 떼는게 가장 좋긴 할텐데 그려려면 다른 작업들이 많아질거같아서 인테페이스만 분리되면 좋을거같습니다

```suggestion
export class RoomsGateway implements IRoomsEventPublisher {

export interface IRoomsEventPublisher {
  roomCreated(room: Room): void
  //... 
}
```

Reply:
혹시 문서화가 필요하시다면 nestjs-asyncapi 가 생각보다 잘 해주더라고요

Reply:
안그래도 노션에 정리하려고 했는데 이런 게 있군요 
알려주셔서 감사해요!